strr-base-web/app/components/BaseUploadAdditionalDocuments.vue

```diff
@@ -0,0 +1,152 @@
+<script setup lang="ts">
+import { v4 as uuidv4 } from 'uuid'
+import type { DefineComponent } from 'vue'
+import type { Form } from '#ui/types'
+const { t } = useI18n()
+const strrModal = useStrrModals()
+const docUploadHelpId = useId() // id for aria-describedby on doc select
+const docFormRef = ref<Form<any>>()
+const showError = ref(false)
+const documentList = ref<UiDocument[]>([])
+
+const props = defineProps<{
+    component: DefineComponent, // either DocumentUploadSelect (Host) or DocumentUploadButton (Strata)
+    applicationNumber: string,
+    isStrata?: boolean, // needed to determine which logic to use
+    selectedDocType: DocumentUploadType | undefined
+}>()
+
+const emit = defineEmits<{
+    uploadDocument: [uiDoc: UiDocument, applicationNumber: string],
+    closeUpload: [void],
+    resetDocType: [void]
+}>()
+
+const addDocumentToList = (doc: File) => {
+  const uiDoc: UiDocument = {
+    file: doc,
+    apiDoc: {} as ApiDocument,
+    name: doc.name,
+    type: props.isStrata ? DocumentUploadType.STRATA_HOTEL_DOCUMENTATION : props.selectedDocType!,
+    id: uuidv4(),
+    loading: false,
+    uploadStep: DocumentUploadStep.NOC,
+    uploadDate: new Date().toISOString().split('T')[0]
+  }
+  documentList.value.push(uiDoc)
+  emit('resetDocType')
+  showError.value = false
+  docFormRef.value?.submit() // submit the form to reset validation
+}
+
+const removeDocumentFromList = (uiDoc: UiDocument) => {
+  const index = documentList.value.findIndex(item => uiDoc.id === item.id)
+  documentList.value.splice(index, 1)
+}
+
+const cancelDocumentsUpload = () => {
+  documentList.value = []
+  emit('resetDocType')
+  emit('closeUpload')
+}
+
+const submitDocuments = async () => {
+  if (documentList.value.length > 0) {
+    for (const doc of documentList.value) {
+      // upload document using docStore's addDocumentToApplication() funtion
+      await emit('uploadDocument', doc, props.applicationNumber)
+    }
+    documentList.value = []
+    emit('closeUpload')
+  } else {
+    showError.value = true
+  }
+}
+
+const handleFileChange = (file: File | File[]) => {
+  if (props.isStrata && Array.isArray(file)) {
+    // if 'component' is Button for strata - array of Files
+    addDocumentToList(file[0]!)
```

Comment:
Non null assertion can be risky if we consider getting an empty array

Reply:
These checks are handled in the 'DocumentUploadSelect' and 'DocumentUploadButton' components that emit the file.